python/pyarrow/fs.py

```diff
@@ -98,9 +98,7 @@ def _filesystem_from_str(uri):
     return filesystem
 
 
-def _ensure_filesystem(
-    filesystem, use_mmap=False, allow_legacy_filesystem=False
-):
+def _ensure_filesystem(filesystem, use_mmap=False):
```

Comment:
Can we make sure those optional args are not passed by position?
```suggestion
def _ensure_filesystem(filesystem, *, use_mmap=False):
```